docs/api/in-app-purchase.md

```diff
@@ -21,10 +21,11 @@ Returns:
 
 The `inAppPurchase` module has the following methods:
 
-### `inAppPurchase.purchaseProduct(productID[, quantity])`
+### `inAppPurchase.purchaseProduct(productID[, quantity, userID])`
```

Comment:
Generally we are now avoiding adding further positional arguments to APIs, as they are hard to read, error-prone, and not forward-compatible. See our [API best practices](https://github.com/electron/governance/blob/main/wg-api/best-practices.md#how-will-this-api-be-extended-in-the-future) guide for more details on that.

In this case where the API existed before these best practices were put into place, the right path forward is a bit unclear. In this case, I think I'd recommend the following to transition to a more future-proof model:

```md
### `inAppPurchase.purchaseProduct(productID[, opts])`

* `productID` string
* `opts` Integer | Object (optional) - If specified as an integer, defines the quantity.
  * `quantity` Integer (optional) - The number of items the user wants to purchase.
  * `username` string (optional) - The string that associates the transaction with a user account on your service (applicationUsername).
```

However, our docs parser (https://github.com/electron/docs-parser) doesn't seem to parse that correctly currently (cc @MarshallOfSound)

Reply:
+1 for that. Do we do a full deprecation cycle of the old API args in that case?

Reply:
i'd probably leave the old api there, it's not too hard to support both (either an int argument or an object argument).

Reply:
I updated the arguments to a new (future-proof) model, but now I get: `Error: The type property is missing on the object provided for typing, something is wrong in the documentation`

Reply:
this is a bug in our docs parser / typing system, as i mentioned in the first comment in the thread :)